composer/loggers/neptune_logger.py

```diff
@@ -0,0 +1,229 @@
+            (default: ``False``).
+        base_namespace (str, optional): The name of the base namespace to log the metadata to. (default: "training")
+        neptune_kwargs (Dict[str, Any], optional): Any additional kwargs to neptune.init_run
+            function. (For help see `Run documentation <https://docs.neptune.ai/api/neptune/#init_run>`_)
+    """
+    METRIC_NAMESPACE = 'metrics'
+    HYPERPARAM_NAMESPACE = 'hyperparameters'
+    TRACE_NAMESPACE = 'traces'
+    INTEGRATION_VERSION_KEY = 'source_code/integrations/neptune-MosaicML'
+
+    def __init__(
+        self,
+        *,
+        project: Optional[str] = None,
+        api_token: Optional[str] = None,
+        rank_zero_only: bool = True,
+        upload_artifacts: bool = False,
+        base_namespace: str = 'training',
+        **neptune_kwargs,
+    ) -> None:
+        try:
+            from neptune import Run
+            from neptune.internal.utils import verify_type
+        except ImportError as e:
+            raise MissingConditionalImportError(extra_deps_group='neptune',
+                                                conda_package='neptune',
+                                                conda_channel='conda-forge') from e
+
+        verify_type('project', project, (str, type(None)))
+        verify_type('api_token', api_token, (str, type(None)))
+        verify_type('rank_zero_only', rank_zero_only, bool)
+        verify_type('upload_artifacts', upload_artifacts, bool)
+        verify_type('base_namespace', base_namespace, str)
+
+        if not base_namespace:
+            raise ValueError("Argument 'base_namespace' cannot be an empty string.")
+
+        self._project = project
+        self._api_token = api_token
+        self._rank_zero_only = rank_zero_only
+        self._upload_artifacts = upload_artifacts
+        self._base_namespace = base_namespace
+        self._neptune_kwargs = neptune_kwargs
+
+        self._enabled = (not rank_zero_only) or dist.get_global_rank() == 0
+        self._neptune_run = None
+        self._base_handler = None
+
+        if self._enabled:
+            self._neptune_run = Run(
+                project=self._project,
+                api_token=self._api_token,
+                **self._neptune_kwargs,
+            )
+            self._base_handler = self._neptune_run[self._base_namespace]
+
+        super(NeptuneLogger, self).__init__()
+
+    @property
+    def neptune_run(self):
+        return self._neptune_run
+
+    @property
+    def base_handler(self):
+        return self._base_handler
+
+    def init(self, state: 'State', logger: 'Logger') -> None:
+        del logger  # unused
+
+        tags = [state.run_name]
+
+        if not self._rank_zero_only:
+            tags.append(f'rank{dist.get_global_rank()}')
+
+        if self._enabled:
+            assert self._neptune_run is not None
+            self._neptune_run['sys/tags'].add(tags)
```

Comment:
We don't need to use tags here

```suggestion
        if not self._rank_zero_only:
            self._base_handler["rank"] = dist.get_global_rank()

        if self._enabled:
            assert self._neptune_run is not None
            self._neptune_run['sys/name'] = state.run_name
```